### ISSUE TITLE
transform function?

### ISSUE THREAD

**Issue by swyxio** [yoshuawuyts/copy-template-dir#13]
I'd love to be able to write my template files in [ejs](https://www.npmjs.com/package/ejs) and just run it through copy-template-dir - what do you think of adding a little transform callback for this?

**Comment by loklaan** [comment]
@sw-yx I like the sound of that, and can appreciate the utility.

I'm unsure, however, if it's an appropriate addition to `copy-template-dir`, as the current "configuration" is done via function arity... and to include an additional cb _after_ the existing `completion` callback would result in an unwieldy api. If you have a suggestion here though that'd differ from adding to the main fn, I'd be super keen to hear!

**Comment by loklaan** [comment]
After I hit the comment button I had another thought...

To make this kind change work (with a nice API) I think a major change for configuration would be preferred; for example it'd be configured by an object instead of fn arity.

This could be done in a backwards compatible way though, by overriding the first arg in the main fn! I'd be happy to support that, since it's fairly cheap imo.

Input from @yoshuawuyts if he has the bandwidth. (_TLDR; we could support more options, such as a transform fn to replace the [maxstache](https://github.com/yoshuawuyts/copy-template-dir/blob/ca7e825e207191c5852248698b646ccc66380c08/index.js#L39) usage.)_

**Comment by swyxio** [comment]
yeah itd be nice to have. up to y'all how you want to execute on it :)

**Comment by swyxio** [comment]
@yoshuawuyts just wanted to say thanks again for this library, its so handy in various things I do!
